### PR TITLE
chore: add reverse to singly linked list

### PR DESCRIPTION
**Pull request by Khushi-Shukla**
[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### Checklist:

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new JavaScript files are placed inside an existing directory.
- [x] All filenames should use the UpperCamelCase (PascalCase) style. There should be no spaces in filenames.
      **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
- [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/JavaScript/pull/1241"></a>

### CODE REVIEW

**Comment by Khushi-Shukla** [comment]
All the tests are passing and the standard style is followed too.

**Review by appgurueu — CHANGES_REQUESTED** [review]
1. How is this a fix?
2. Please integrate this into https://github.com/TheAlgorithms/JavaScript/blob/master/Data-Structures/Linked-List/SinglyLinkedList.js; this definitely should not be its own class.

**Comment by Khushi-Shukla** [comment]
Please review the changes. Function added in SinglyLinkedList.js and tested too

File: Data-Structures/Linked-List/SinglyLinkedList.js [hunk 1]
```diff
@@ -274,6 +274,20 @@ class LinkedList {
   log () {
     console.log(JSON.stringify(this.headNode, null, 2))
   }
+
+  // Method to reverse the LinkedList where each node contains a single integer value
```
Comment (appgurueu) [inline Data-Structures/Linked-List/SinglyLinkedList.js:278]: It doesn't matter what the nodes contain

File: Data-Structures/Linked-List/SinglyLinkedList.js [hunk 2]
```diff
@@ -274,6 +274,20 @@ class LinkedList {
...
+  reverseList () {
```
Comment (appgurueu) [inline Data-Structures/Linked-List/SinglyLinkedList.js:279]: You may omit the redundant `List` from the method name.

File: Data-Structures/Linked-List/test/SinglyLinkedList.test.js [hunk 3]
```diff
@@ -247,4 +247,10 @@ describe('SinglyLinkedList', () => {
     headNode.rotateListRight(5)
     expect(headNode.get()).toEqual([20, 30, 40, 50, 10])
   })
+
+  it('Reverse a number represented as a Linked List where each node has a single integer', () => {
+    const headNode = new LinkedList([4, 3, 1])
```
Comment (appgurueu) [inline Data-Structures/Linked-List/test/SinglyLinkedList.test.js:252]: This is not a headNode but rather a list object

Comment (Khushi-Shukla) [reply Data-Structures/Linked-List/test/SinglyLinkedList.test.js]: Changed it to list.

File: Data-Structures/Linked-List/SinglyLinkedList.js (hunk 2 above)
Comment (Khushi-Shukla) [reply Data-Structures/Linked-List/SinglyLinkedList.js]: Done.

File: Data-Structures/Linked-List/SinglyLinkedList.js (hunk 1 above)
Comment (Khushi-Shukla) [reply Data-Structures/Linked-List/SinglyLinkedList.js]: Corrected.

**Review by appgurueu — APPROVED** [review]

**Comment by Khushi-Shukla** [comment]
@raklaptudirm Please review this PR

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**